src/frameworks/flutter/utils.ts

```diff
@@ -8,3 +10,27 @@ export function assertFlutterCliExists() {
       "Flutter CLI not found, follow the instructions here https://docs.flutter.dev/get-started/install before trying again.",
     );
 }
+
+export async function getTreeShakeFlag(): Promise<string> {
+  const pubSpecPath = "./pubspec.yaml";
+  let pubSpec: Record<string, any> = {};
+  try {
+    const pubSpecBuffer = await readFile(pubSpecPath);
+    pubSpec = yaml.parse(pubSpecBuffer.toString());
+  } catch (error) {
+    console.info("pubspec.yaml not found, skipping tree shaking");
+    return "";
+  }
+
+  const treeShakePackages = [
+    "material_icons_named",
+    "material_symbols_icons",
+    "material_design_icons_flutter",
+    "flutter_iconpicker",
+    "font_awesome_flutter",
+    "ionicons_named",
+  ];
```

Comment:
can you document why we need this? Probably linking to the issue #6197 is good enough